testing/trino-product-tests/src/main/java/io/trino/tests/jdbc/TestExternalAuthorizerOAuth2.java

```diff
@@ -80,13 +80,36 @@ public void shouldAuthenticateAndExecuteQuery()
         Properties properties = new Properties();
         properties.setProperty("user", "test");
         try (Connection connection = DriverManager.getConnection(jdbcUrl, properties);
-                PreparedStatement statement = connection.prepareStatement("select * from tpch.tiny.nation");
+                PreparedStatement statement = connection.prepareStatement("SELECT * FROM tpch.tiny.nation");
                 ResultSet results = statement.executeQuery()) {
             assertThatNoException().isThrownBy(() -> checkLastLoginActionExecutionFailures());
             assertThat(forResultSet(results)).matches(TpchTableResults.PRESTO_NATION_RESULT);
         }
     }
 
+    @Test(groups = {OAUTH2, PROFILE_SPECIFIC_TESTS})
+    @Flaky(issue = "https://github.com/trinodb/trino/issues/6991", match = "Last login action has failed with exception")
+    public void shouldAuthenticateAfterTokenExpires()
+            throws Exception
+    {
+        prepareSeleniumHandler();
+        Properties properties = new Properties();
+        properties.setProperty("user", "test");
+        try (Connection connection = DriverManager.getConnection(jdbcUrl, properties);
+                PreparedStatement statement = connection.prepareStatement("SELECT * FROM tpch.tiny.nation");
+                ResultSet results = statement.executeQuery()) {
+            assertThatNoException().isThrownBy(() -> checkLastLoginActionExecutionFailures());
+            //Wait until the token expires. See: HydraIdentityProvider.TTL_ACCESS_TOKEN_IN_SECONDS
+            SECONDS.sleep(10);
+
+            try (PreparedStatement repeatedStatement = connection.prepareStatement("SELECT * FROM tpch.tiny.nation");
+                    ResultSet repeatedResults = repeatedStatement.executeQuery()) {
+                assertThatNoException().isThrownBy(() -> checkLastLoginActionExecutionFailures());
```

Comment:
It too me a while to understand this assertion.

Maybe just (without `assertThatNoException`):
```
checkLastLoginActionExecutionFailures();
```

This comment applies to other `assertThatNoException` usages.

Also I would rename this to `assertSuccessfulLogin`.